shared/actions/crypto.tsx

```diff
@@ -72,6 +68,19 @@ const handleSaltpackOpenFile = (action: CryptoGen.OnSaltpackOpenFilePayload) =>
   })
 }
 
+// Mobile is split into two routes (input and output). This Saga handler
+// transitions to the output route on success
+const handleMobileOperationSuccess = (action: CryptoGen.OnOperationSuccessPayload) => {
+  if (!Platform.isMobile) return
```

Comment:
i think we prefer the components to be doing the nav in router v2 land
also if you're going to bail on non mobile you can just not hook up the side effect below

Reply:
1. So my components should be using `useSelector`, looking for the success action, and routing as a effect?

2. How do you mean not hook up the side effect below?

Reply:
1. if its not a pain but thats the direction we're going. component is 'waiting' and it gets a success so it navs to the next thing
2. i mean you can just not do the chainAction aka
```
if (Platform.isMobile) {
yield * Saga.chainAction2...
}
```

Reply:
Yeah it seems doable to do this routing in a native component. I can move all of this there.